src/libs/ReportActionsUtils.ts

```diff
@@ -631,6 +653,121 @@ function isNotifiableReportAction(reportAction: OnyxEntry<ReportAction>): boolea
 }
 
+function getChannelLogMemberAction(reportAction: OnyxEntry<ReportAction>) {
+    const messageItems: MessageActionItemChanelLog[] = [];
+    const isInviteRoom = isInvitedRoom(reportAction);
+
+    const verb = isInviteRoom ? Localize.translateLocal('workspace.invite.invited') : Localize.translateLocal('workspace.invite.removed');
+
+    messageItems.push({
+        kind: 'text',
+        content: `${verb} `,
+    });
+
+    const originalMessage = reportAction?.originalMessage as CommonOriginalMessage;
+    const targetAccountIDs: number[] = originalMessage?.targetAccountIDs ?? [];
+    const mentions = targetAccountIDs.map((accountID) => {
+        const personalDetail = allPersonalDetails?.[accountID];
+
+        if (personalDetail) {
+            const displayNameOrLogin =
+                LocalePhoneNumber.formatPhoneNumber(personalDetail.login ?? '') || personalDetail?.displayName ? personalDetail?.displayName : Localize.translateLocal('common.hidden');
+            return {content: `@${displayNameOrLogin}`, accountID};
+        }
+        return {content: '', accountID: 0};
+    });
+
+    const lastMention = mentions.pop();
+
+    if (mentions.length === 0) {
+        messageItems.push({
+            kind: 'userMention',
+            ...lastMention,
+        });
+    } else if (mentions.length === 1) {
+        messageItems.push(
+            {
+                kind: 'userMention',
+                ...mentions[0],
+            },
+            {
+                kind: 'text',
+                content: ` ${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                ...lastMention,
+            },
+        );
+    } else {
+        mentions.forEach((mention) => {
+            messageItems.push({
+                kind: 'userMention',
+                ...mention,
+                content: `${mention.content}, `,
+                isComma: true,
+            });
+        });
+
+        messageItems.push(
+            {
+                kind: 'text',
+                content: `${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                ...lastMention,
+            },
+        );
+    }
+
+    const roomName = originalMessage?.roomName;
+    if (roomName) {
+        const preposition = isInviteRoom ? ` ${Localize.translateLocal('workspace.invite.to')} ` : ` ${Localize.translateLocal('workspace.invite.from')} `;
+
+        messageItems.push(
+            {
+                kind: 'text',
+                content: preposition,
+            },
+            {
+                kind: 'roomReference',
+                roomName,
+                roomID: originalMessage?.reportID,
+                content: roomName,
+            },
+        );
+    }
+
+    return messageItems;
+}
+
+function getActionItemFragmentChanelLog(reportAction: OnyxEntry<ReportAction>) {
```

Comment:
I think the suitable name would be `getActionItemFragmentChanelLogFragment` (with _Fragment_ suffix), what do you think? We could also try typing the return value, especially if we have a ready TypeScirpt type already sitting somewhere.